### PR TITLE
chore: update consignments modal headers

### PR DESCRIPTION
**Pull request by brainbicycle**
The type of this PR is: **chore**

This PR resolves [CX-1183]

### Description

Uses the FancyModalHeader component in all screens in the consignments flow and adjusts some top margins. Also deletes some unused code. 

<details>
<summary>Android</summary>

Screenshot_1617042031

Screenshot_1617042133

Screenshot_1617042250

Screenshot_1617042258

Screenshot_1617042268

Screenshot_1617042277

Screenshot_1617042285

</details>

<details>
<summary>iOS 14</summary>

Simulator Screen Shot - iPhone 11 Pro Max - 2021-03-29 at 14 14 27

Simulator Screen Shot - iPhone 11 Pro Max - 2021-03-29 at 14 14 40

Simulator Screen Shot - iPhone 11 Pro Max - 2021-03-29 at 14 14 48

Simulator Screen Shot - iPhone 11 Pro Max - 2021-03-29 at 14 14 54

Simulator Screen Shot - iPhone 11 Pro Max - 2021-03-29 at 14 15 00

Simulator Screen Shot - iPhone 11 Pro Max - 2021-03-29 at 14 15 05

Simulator Screen Shot - iPhone 11 Pro Max - 2021-03-29 at 14 15 19

</details>

<details>
<summary>iOS 12</summary>

Simulator Screen Shot - iPhone X (12 4) - 2021-03-29 at 14 17 16

Simulator Screen Shot - iPhone X (12 4) - 2021-03-29 at 14 17 20

Simulator Screen Shot - iPhone X (12 4) - 2021-03-29 at 14 17 26

Simulator Screen Shot - iPhone X (12 4) - 2021-03-29 at 14 17 32

Simulator Screen Shot - iPhone X (12 4) - 2021-03-29 at 14 17 40

Simulator Screen Shot - iPhone X (12 4) - 2021-03-29 at 14 17 46

Simulator Screen Shot - iPhone X (12 4) - 2021-03-29 at 14 17 59

</details>

<details>
<summary>iOS 12 - Confirmation Flow</summary>

Simulator Screen Shot - iPhone X (12 4) - 2021-03-30 at 09 02 43

Simulator Screen Shot - iPhone X (12 4) - 2021-03-30 at 09 04 08

Simulator Screen Shot - iPhone X (12 4) - 2021-03-30 at 09 10 09

</details>

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-1183]: https://artsyproduct.atlassian.net/browse/CX-1183

### CODE REVIEW

File: src/lib/Scenes/Consignments/Screens/Overview.tsx [hunk 1]
```diff
@@ -249,8 +250,7 @@ export class Overview extends React.Component<Props, State> {
       this.state.metadata.height &&
       this.state.metadata.width &&
       this.state.editionScreenViewed &&
-      // TODO: make photos work on Android
-      (Platform.OS === "android" || this.state.photos?.length! /* STRICTNESS_MIGRATION */ > 0)
+      this.state.photos?.length! > 0
```
Comment (brainbicycle) [inline src/lib/Scenes/Consignments/Screens/Overview.tsx:253]: I forgot to remove this when adding support for photo picker on Android

**Review by ds300 — APPROVED** [review]
Great work! I found one tiny UI thing to fix.

I also noticed that you didn't add any toolbar headers for the phone confirmation step or the final submission status screen.

We should check with @samchieng  but I'd propose to add a blank toolbar with an x button on both screens to allow the user to dismiss the modal in the standard way. We do that in auction bid/registration and I think it'd be good to do the same thing here.

**Comment by brainbicycle** [comment]
> Great work! I found one tiny UI thing to fix.
> 
> I also noticed that you didn't add any toolbar headers for the phone confirmation step or the final submission status screen.
> 
> We should check with @samchieng but I'd propose to add a blank toolbar with an x button on both screens to allow the user to dismiss the modal in the standard way. We do that in auction bid/registration and I think it'd be good to do the same thing here.

Good catch, I added there as well using @admbtlr confirmation alert to avoid quitting in progress submissions. @admbtlr I had to move function out of useFocus hook let me know if it looks okay!

File: src/lib/Scenes/Consignments/Screens/ConfirmContactInfo.tsx [hunk 2]
```diff
@@ -36,24 +36,24 @@ const ConfirmContactInfo: React.FC<{
...
-            { text: "Leave Now", onPress: () => dismissModal() },
-            { text: "Wait", style: "default" },
-          ])
-        } else {
-          dismissModal()
-        }
-        return true
-      }
-
-      BackHandler.addEventListener("hardwareBackPress", onBackPress)
```
[the remark this answers is not in the document]
Comment (admbtlr) [reply src/lib/Scenes/Consignments/Screens/ConfirmContactInfo.tsx:51]: Yes, that looks much better!

**Pull request merged**